system/Test/CIUnitTestCase.php

```diff
@@ -185,16 +192,14 @@ protected function setUp(): void
 			$this->$method();
 		}
 
-		// Check for trait methods
-		foreach (class_uses_recursive($this) as $trait)
+		// Check for the database trait
+		if (method_exists($this, 'setUpDatabase'))
 		{
-			$method = 'setUp' . class_basename($trait);
-
-			if (method_exists($this, $method))
-			{
-				$this->$method();
-			}
+			$this->setUpDatabase();
 		}
+
+		// Check for other trait methods
+		$this->callTraitMethods('setUp');
```

Comment:
Wouldn't this end up running the database methods twice?

Reply:
Nope, because I renamed its method to `setUpDatabase()` which no longer matches the paradigm `setUp{TraitName}()`. Good catch though, I almost made that mistake myself!